File: fibonacci.cpp
#include <iostream>
#include <vector>

using namespace std;

int fib (int n)
{
    if (n == 0 || n == 1)
    {
        return n;
    }
    else
    {
        return fib(n - 1) + fib (n - 2);
    }
}

void display (int n)
{
    for (int i = 0; i < n; i ++)
    {
        cout << fib(i) << " ";
    }
}

int main ()
{
    cout << "Enter the position you want from fibonacci : ";
    int n;
    cin >> n;

    cout << "Number at position " << n << " in Fibonacci series  : ";
    cout << fib(n - 1);
    cout << endl;

    cout << "Enter the limit of the series : ";
    cin >> n;
    cout << endl;
    cout << "Fibonacci series upto position " << n << " is : ";
    display(n);

    return 0;
}

File: Fibonacci.java
import java.util.Scanner;

public class Fibonacci
{
    public static void main(String args[])
    {
        Scanner s = new Scanner(System.in);
        System.out.println("Enter the limit of fibonacci series : ");
        int num = s.nextInt();
        System.out.println("Fibonacci series upto entered limt is : ");
        fib(num);
        s.close();
    }
    static void fib(int n)
    {
        int a = 0;
        int b = 1;
        System.out.print(a + " " + b);
        while(n -2 > 0)
        {
            int c = a + b;
            a = b;
            b = c;
            System.out.print(" " + c);
            n--;
        }
    }
}